exercises/saddle-points/example.rs

```diff
@@ -1,23 +1,24 @@
 pub fn find_saddle_points(input: &[Vec<u64>]) -> Vec<(usize, usize)> {
-    let mut saddle_points = Vec::new();
+    let (width, height) = match (input.len(), input[0].len()) {
```

Comment:
This isn't actually safe, anyway. Try adding this test:

```rust
#[test]
fn empty() {
    let input = vec![];
    let expected: Vec<(usize, usize)> = Vec::new();
    assert_eq!(expected, find_saddle_points(&input));
}
```

When I run that test on this code, I get

```
failures:

---- empty stdout ----
thread 'empty' panicked at 'index out of bounds: the len is 0 but the index is 0', src/lib.rs:2:47
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace.


failures:
    empty
```

That's because the `match` implementation isn't clever enough to short-circuit and avoid the second calculation if the first is sufficient to determine a branch, so it unconditionally attempts `input[0]` as well.

Reply:
That's a serious overlook. Then https://github.com/exercism/rust/pull/910#discussion_r354475074 gets even more relevant.